sem/utils.py

```diff
@@ -573,39 +581,39 @@ def parse_logs(log_file):
                 continue
 
             # If severity class is 'function', then arguments cannot be empty
-            if groups[7] is None and groups[8] is None and groups[6] is None:
+            if groups.group('severity_class') is None and groups.group('message') is None and groups.group('arguments') is None:
                 warnings.warn("Log format is not consistent with prefix_all. Skipping log '%s'" % log, RuntimeWarning)
                 continue
 
             temp_dict = None
             # Remove trailing whitespaces after message
-            message = groups[8]
-            if groups[8] is not None:
+            message = groups.group('message')
+            if message is not None:
                 message = message.rstrip()
 
             # If level is function
             # TODO - I have seen in certain examples that the format of
             # level=function is different.
-            if groups[7] is None and groups[8] is None:
+            if groups.group('severity_class') is None and groups.group('message') is None:
                 temp_dict = {
-                    'Time': float(groups[1]),
-                    'Context': groups[2],
-                    'Extended_context': groups[3],
-                    'Component': groups[4],
-                    'Function': groups[5],
-                    'Arguments': groups[6],
+                    'Time': float(groups.group('time')),
+                    'Context': groups.group('context'),
+                    'Extended_context': groups.group('extended_context'),
+                    'Component': groups.group('component'),
+                    'Function': groups.group('function'),
+                    'Arguments': groups.group('arguments'),
                     'Severity_class': 'FUNCTION',
                     'Message': ''
                 }
             else:
                 temp_dict = {
-                    'Time': float(groups[1]),
-                    'Context': groups[2],
-                    'Extended_context': groups[3],
-                    'Component': groups[4],
-                    'Function': groups[5],
-                    'Arguments': groups[6],
-                    'Severity_class': groups[7],
+                    'Time': float(groups.group('time')),
+                    'Context': groups.group('context'),
+                    'Extended_context': groups.group('extended_context'),
+                    'Component': groups.group('component'),
+                    'Function': groups.group('function'),
+                    'Arguments': groups.group('arguments'),
+                    'Severity_class': groups.group('severity_class'),
                     'Message': message
```

Comment:
i still believe that dictionary keys should be in lower case

Reply:
Fixed in [df1ece8](https://github.com/signetlabdei/sem/pull/51/commits/df1ece8ea8cc4ce1c3a3c5ea67d101fce2b060c5).